build.gradle

```diff
@@ -34,6 +35,7 @@ dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	implementation 'pentaho-kettle:kettle-core:8.2.0.7-719'
 	implementation 'pentaho-kettle:kettle-engine:8.2.0.7-719'
+	pdiPlugin 'org.pentaho.di.plugins:pdi-core-plugins-impl:8.2.0.7-719'
```

Comment:
this seems to be available from maven, https://javalibs.com/artifact/org.pentaho.di.plugins/pdi-core-plugins-impl is this the way u load it from there! BTW what reason is behind copying it into plugins/? what are we planning to use this folder for?

Reply:
I tried to explain it on 36fa4cf. 

`kettle-core` does not include all types of steps that you can use in `spoon`. 

Some steps are defined in plugins that are loaded at runtime by kettle. It seems that the only way to load this plugins is to store them in a folder that kettle will scan when starting (no classpath scanning).

[You can either use `./plugins`, `~/.kettle/plugins` or define system property `KETTLE_PLUGIN_BASE_FOLDERS`](https://github.com/pentaho/pentaho-kettle/blob/0cd5339daeca9baafb72c26ea1ea5dd98174e884/core/src/main/java/org/pentaho/di/core/plugins/PluginFolder.java#L97-L100).

The [Abort](https://github.com/pentaho/pentaho-kettle/blob/8.2.0.7-R/plugins/core/impl/src/main/java/org/pentaho/di/trans/steps/abort/Abort.java) used in the validations transformation is not included in core.

I think the best way to handle this situation is to have a custom gradle configuration for separating the scope of this plugin dependencies and to rely on `./plugins` to have kettle load them. That is the reason for `pdiPlugin`, the copy step before build and extracting the jar to `plugins/pdi-core-plugins-impl-8.2.0.7-719.jar`.

Maybe there is a better way to handle this.